modules/userId/index.js

```diff
@@ -969,15 +989,26 @@ function canUseCookies(submodule) {
   return true
 }
 
+function calculateEnabledStorageTypes(submodule) {
+  const storageTypes = getConfiguredStorageTypes(submodule.config);
+
+  return storageTypes.filter(type => {
+    if (type === LOCAL_STORAGE) {
+      return canUseLocalStorage(submodule);
+    } else if (type === COOKIE) {
+      return canUseCookies(submodule);
+    }
+
+    return false;
+  });
+}
+
 function canUseStorage(submodule) {
-  const storageType = submodule?.config?.storage?.type;
-  if (storageType === COOKIE) {
-    return canUseCookies(submodule);
-  } else if (storageType === LOCAL_STORAGE) {
-    return canUseLocalStorage(submodule);
-  }
+  const enabledStorageTypes = calculateEnabledStorageTypes(submodule);
+
+  submodule.enabledStorageTypes = enabledStorageTypes;
```

Comment:
The function name `canUseStorage` implies that it's only querying something, but setting `enabledStoreTypes` here mutates the submodule state. I'd move this line up to the caller, even if it makes the caller somewhat more complicated.